Marlin/Conditionals_post.h

```diff
@@ -513,7 +513,7 @@
   #define HAS_E3_STEP (PIN_EXISTS(E3_STEP))
   #define HAS_E4_STEP (PIN_EXISTS(E4_STEP))
   #define HAS_DIGIPOTSS (PIN_EXISTS(DIGIPOTSS))
-  #define HAS_BUZZER (PIN_EXISTS(BEEPER) || ENABLED(LCD_USE_I2C_BUZZER))
+  #define HAS_BUZZER (ENABLED(ULTRA_LCD) && (PIN_EXISTS(BEEPER) || ENABLED(LCD_USE_I2C_BUZZER)))
 
```

Comment:
Yes, I had noticed that. But then it seemed like it might be passing-through because some boards also define encoder buttons without an LCD controller.

So… I've added a `buzz(f,d)` function to `Marlin_main.cpp` that calls `lcd_buzz` if there's an LCD with an i2c buzzer, or calls `buzzer.tone` if there's a `BEEPER_PIN`.